source/Plugins.cpp

```diff
@@ -0,0 +1,133 @@
+/* Plugins.cpp
+Copyright (c) 2022 by Sam Gleske (samrocketman on GitHub)
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "Plugins.h"
+
+#include "DataFile.h"
+#include "DataNode.h"
+#include "DataWriter.h"
+#include "Files.h"
+
+#include <algorithm>
+#include <map>
+
+using namespace std;
+
+namespace {
+	map<string, bool> settings;
+	map<string, bool> frozenSettings;
+
+	void LoadSettings(const string &path)
+	{
+		DataFile prefs(path);
+		for(const DataNode &node : prefs)
+		{
+			const string &key = node.Token(0);
+			if(key == "plugins")
+				for(const DataNode &child : node)
+					if(child.Size() == 2)
+						settings[child.Token(0)] = child.Value(1);
+		}
+	}
+}
+
+
+
+void Plugins::Load()
+{
+	// Global plugin settings
+	LoadSettings(Files::Resources() + "plugins.txt");
+	// Local plugin settings
+	LoadSettings(Files::Config() + "plugins.txt");
+}
+
+
+
+void Plugins::Save()
+{
+	if(settings.empty())
+		return;
+	DataWriter out(Files::Config() + "plugins.txt");
+
+	out.Write("plugins");
+	out.BeginChild();
+	{
+		for(const auto &it : settings)
+			out.Write(it.first, it.second);
+	}
+	out.EndChild();
+}
+
+
+
+// Freeze the plugin set to determine if a setting has changed.
+void Plugins::Freeze()
+{
+	copy(settings.begin(), settings.end(), inserter(frozenSettings, frozenSettings.end()));
```

Comment:
I’m not sure I understand; do you mean with a loop and do

```
for(…)
{
    frozenSettings[key] = settingsValue;
}
```

As a pseudo example?

Reply:
I created a copy because I didn't want changes in settings to also affect frozenSettings.  I can do some debugging to verify if `frozenSettings = settings` is good enough.  Would be good if @tehhowch can confirm that's what they mean.

Reply:
That's exactly what I mean.

Reply:
Ok I will try with const as well.  It sounds easy

Reply:
The implementation has changed and copying the map is no longer used.  Instead a boolean pair.  Please see the latest diff for `Plugins::Freeze()` method.  Does this work to resolve feedback?